utils/imagenet_topk_accuracy_driver.py

```diff
@@ -0,0 +1,318 @@
+        curr_label_idx = curr_label_idx + 1
+    return img_paths, img_labels
+
+## Given an image located at @param img_path, transform the image
+# and save it to the path @param path_to_new_img.
+def resize_and_save_image(img_path, path_to_new_img):
+    # Load the image.
+    img = default_image_loader(img_path)
+
+    # Use to Resize and CenterCrop the images to 224x224.
+    transform_resize = transforms.Compose([
+        transforms.Resize(256),
+        transforms.CenterCrop(224),
+    ])
+
+    resized_img = transform_resize(img)
+    resized_img.save(path_to_new_img, format="png")
+
+## Used to pre-process an input set of images. Takes a string of a directory
+# @param validation_images_dir and saves the cropped subset of the images in a
+# subdirectory `processed/`, which must not yet exist.
+def save_centered_cropped_dataset(validation_images_dir):
+    assert os.path.exists(validation_images_dir), "Validation directory does not exist: " + validation_images_dir
+
+    processed_validation_images_dir = os.path.join(validation_images_dir,
+                                                   "processed")
+    print("Saving centered cropped input images:" +
+          processed_validation_images_dir)
+
+    img_dirs = []
+    for img_dir in os.listdir(validation_images_dir):
+        dir_path = os.path.join(validation_images_dir, img_dir)
+        if os.path.isdir(dir_path):
+            img_dirs.append(img_dir)
+    img_dirs.sort()
+
+    try:
+        os.makedirs(processed_validation_images_dir)
+    except OSError:
+        sys.exit("New validation directory must not exist")
+
+    # Iterate over all labels subdirectories, loading, transforming and saving
+    # all images to the new location.
+    for img_dir in img_dirs:
+        orig_dir_path = os.path.join(validation_images_dir, img_dir)
+        processed_dir_path = os.path.join(processed_validation_images_dir, img_dir)
+
+        # Create a new subdirectory for the next label.
+        try:
+            os.makedirs(processed_dir_path)
+        except OSError:
+            sys.exit("New label subdirectory somehow already existed.")
+
+        # Transform and save all images in this label subdirectory.
+        for orig_img_filename in os.listdir(orig_dir_path):
+            orig_img_path = os.path.join(orig_dir_path, orig_img_filename)
+            if os.path.isfile(orig_img_path):
+                processed_img_path = os.path.join(processed_dir_path,
+                                                  orig_img_filename)
+                resize_and_save_image(orig_img_path, processed_img_path)
+
+## @returns a list of strings (of length equal to the batch size) which are
+# paths to images to do inference on. @param img_paths is the set of all image
+# paths, @param img_index is the next index to use in @param img_paths, and
+# @param tmp_dir_name is the location of where to save the images if necessary.
+def get_curr_img_paths(img_paths, img_index, tmp_dir_name, batch_size,
```

Comment:
I see there is a lot of duplicated logic between the function above and this one. Could you do "crop and center" for all images together before starting to run image-classifier? That way, all this logic can go into one function. It can even cache automatically by looking at whether "processed" dir already exists.